src/main/kotlin/notary/endpoint/eth/IrohaRefundStrategyImpl.kt

```diff
@@ -0,0 +1,53 @@
+package notary.endpoint.eth
+
+import com.github.kittinunf.result.map
+import model.IrohaCredential
+import mu.KLogging
+import sidechain.iroha.consumer.IrohaConsumer
+import sidechain.iroha.consumer.IrohaNetwork
+import sidechain.iroha.util.ModelUtil
+import wdrollbackservice.WdRollbackServiceImpl
+
+class IrohaRefundStrategyImpl(
+    private val irohaNetwork: IrohaNetwork,
+    private val credential: IrohaCredential,
+    private val notaryAccountId: String,
+    private val irohaConsumer: IrohaConsumer
+) : IrohaRefundStrategy {
+
+    override fun performRefund(request: IrohaRefundRequest): IrohaNotaryResponse {
+        val irohaTransaction = ModelUtil.getTransaction(irohaNetwork, credential, request.irohaTx)
+
+        return irohaTransaction
+            .map { tx ->
+                tx.payload.reducedPayload.commandsList.first { command ->
+                    val transferAsset = command.transferAsset
+                    transferAsset?.srcAccountId != "" && transferAsset?.destAccountId == notaryAccountId
+                }
+            }
+            .map { transferCommand ->
+                val destAccountId = transferCommand?.transferAsset?.srcAccountId
+                        ?: throw IllegalStateException("Unable to identify primary Iroha transaction data")
+
+                try {
+                    ModelUtil.transferAssetIroha(
+                        irohaConsumer,
+                        notaryAccountId,
+                        destAccountId,
+                        transferCommand.transferAsset.assetId,
+                        "Rollback transaction due to failed withdrawal in Ethereum",
+                        transferCommand.transferAsset.amount
+                        // TODO: Implement quorum
+                    ).map {
+                        logger.info(it)
+                        IrohaNotaryResponse.Successful(it)
+                    }.get()
+                } catch (ex: Exception) {
+                    WdRollbackServiceImpl.logger.error("Error during rollback tx", ex)
```

Comment:
This logger is taken from another file by mistake

Reply:
Done